vector/array.h

```diff
@@ -0,0 +1,26 @@
+#ifndef array_h
+#define array_h
```

Comment:
[Include guards](https://en.wikipedia.org/wiki/Include_guard), nice.

However, they have a possibility of **name clashing**. If you happen to have two headers with the same include guard, you will end up with weird compilation results which may take you some time till you understand what is going on (well `-E` gcc flag would show what is going on - or rather what have been included and what not, but it might not come up to one's mind).

Also, there is a [convention to write macros UPPERCASE](http://stackoverflow.com/questions/3799478/c-ifndef-for-include-files-why-is-all-caps-used-for-the-header-file), so such ifndef/define's are usually written uppercase as well.


There is another solution to the problem of including a file multiple times, which is **pragma once**. This is **NON-STANDARD** solution, but most of the compilers actually support it.

Read more here:
* https://en.wikipedia.org/wiki/Pragma_once
* http://stackoverflow.com/questions/1143936/pragma-once-vs-include-guards